src/vaults/LockedWithdrawal.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+
+import {AffineVault} from "src/vaults/AffineVault.sol";
+
+contract LockedWithdrawalEscrow is ERC20 {
+    using SafeTransferLib for ERC20;
+    using FixedPointMathLib for uint256;
+
+    // token paid to user
+    ERC20 immutable asset;
+
+    // Last withdrawal request time map
+    mapping(address => uint256) requestTimes;
+
+    // amount of pending debt token share to resolve
+    // The share assigned to the user immediately, need to keep track of it until resolved.
+    uint256 pendingDebtToken;
+
+    // Max locked withdrawal time, user can withdaraw funds after sla
+    uint256 immutable sla;
+
+    AffineVault vault;
+
+    constructor(AffineVault _vault, uint256 _sla) ERC20("DebtToken", "DT", 18) {
+        asset = ERC20(_vault.asset());
+        vault = _vault;
+        sla = _sla;
+    }
+
+    /**
+     * @notice User register to withdraw earn token
+     * @param user user address
+     * @param debtShare amount of debt token share for the withdrawal request
+     * NB: user withdrawal request will be locked until the SLA time is over.
+     * NB: debtTokenShare = token_to_withdraw * price of token
+     */
+    function registerWithdrawalRequest(address user, uint256 debtShare) external {
+        // check if the sender is valut
+        require(address(vault) == msg.sender, "Unrecognized vault");
```

Comment:
We generally put the contract name in the revert message. See [here](https://github.com/AffineLabs/contracts/blob/7f8b94bd8f56c7e451eb24247e36636032848f53/src/vaults/cross-chain-vault/BaseVault.sol#L140). 

`LW: must be vault` would work.